src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/ProcessTests.cs

```diff
@@ -145,7 +145,137 @@ public async Task MultiProcessIdentificationTest(DiagnosticPortConnectionMode mo
+#if NET5_0_OR_GREATER
+        [InlineData(DiagnosticPortConnectionMode.Listen)]
+#endif
+        public async Task ProcessRequestConsistencyTest(DiagnosticPortConnectionMode mode)
+        {
+            DiagnosticPortHelper.Generate(
+                mode,
+                out DiagnosticPortConnectionMode appConnectionMode,
+                out string diagnosticPortPath);
+
+            await using MonitorRunner toolRunner = new(_outputHelper);
+            toolRunner.ConnectionMode = mode;
+            toolRunner.DiagnosticPortPath = diagnosticPortPath;
+            toolRunner.DisableAuthentication = true;
+            await toolRunner.StartAsync();
+
+            using HttpClient httpClient = await toolRunner.CreateHttpClientDefaultAddressAsync(_httpClientFactory);
+            ApiClient apiClient = new(_outputHelper, httpClient);
+
+            const int appCount = 1; // May add more extensive testing in the future that utilizes multiple processes
+            AppRunner[] appRunners = new AppRunner[appCount];
+
+            for (int i = 0; i < appCount; i++)
+            {
+                AppRunner runner = new(_outputHelper, Assembly.GetExecutingAssembly(), appId: i + 1);
+                runner.ConnectionMode = appConnectionMode;
+                runner.DiagnosticPortPath = diagnosticPortPath;
+                runner.ScenarioName = TestAppScenarios.AsyncWait.Name;
+                runner.Environment[ExpectedEnvVarName] = Guid.NewGuid().ToString("D");
+
+                appRunners[i] = runner;
+            }
+
+            IEnumerable<ProcessIdentifier> identifiers;
+            await appRunners.ExecuteAsync(async () =>
+            {
+                // Query for process identifiers
+                identifiers = await apiClient.GetProcessesAsync();
+
+                Assert.NotNull(identifiers);
+
+                List<int?> pids = new List<int?>();
+                List<Guid?> uids = new List<Guid?>();
+                List<string> names = new List<string>();
+
+                foreach (ProcessIdentifier processIdentifier in identifiers)
+                {
+                    pids.Add(processIdentifier.Pid);
+                    uids.Add(processIdentifier.Uid);
+                    names.Add(processIdentifier.Name);
+                }
+
+#if NET5_0_OR_GREATER
+                // CHECK 1: Get response for processes using PID, UID, and Name and check for consistency
+
+                List<ProcessInfo> processInfoQueriesCheck1 = new List<ProcessInfo>();
+
+                processInfoQueriesCheck1.Add(await apiClient.GetProcessAsync(pid: pids[0], uid: null, name: null, System.Threading.CancellationToken.None));
+                processInfoQueriesCheck1.Add(await apiClient.GetProcessAsync(pid: null, uid: uids[0], name: null, System.Threading.CancellationToken.None));
+                //processInfoQueriesCheck1.Add(await apiClient.GetProcessAsync(pid: null, uid: null, name: names[0], System.Threading.CancellationToken.None));
```

Comment:
I don't think we'll ever only check by name in the unit tests as they are because the process is always `dotnet` regardless of which unit test is being run.